distributed/shuffle/_scheduler_plugin.py

```diff
@@ -383,8 +383,28 @@ def transition(
         **kwargs: Any,
     ) -> None:
         """Clean up scheduler and worker state once a shuffle becomes inactive."""
-        if finish not in ("released", "forgotten"):
+        if finish not in ("released", "erred", "forgotten"):
             return
+
+        if finish == "erred":
+            ts = self.scheduler.tasks[key]
+            for active_shuffle in self.active_shuffles.values():
+                barrier = self.scheduler.tasks[barrier_key(active_shuffle.id)]
+                if (
+                    ts == barrier
+                    or ts in barrier.dependents
+                    or ts in barrier.dependencies
+                ):
```

Comment:
This check might become expensive for large graphs with many independent P2P operations. I think I am fine with this for now since it only triggers if some task failed. Should this become a problem, we can archive the shuffle at the same time as we log the event. This will add more complexity though, so I haven't done it yet.

Reply:
Just checking something: This plugin is hit for every computation not only P2P shuffles, correct?

Reply:
Yes, this hook is hit for _every single transition_.